virtual_rainforest/models/litter/litter_model.py

```diff
@@ -0,0 +1,178 @@
+
+from __future__ import annotations
+
+from typing import Any
+
+import numpy as np
+from pint import Quantity
+
+from virtual_rainforest.core.base_model import BaseModel
+from virtual_rainforest.core.data import Data
+from virtual_rainforest.core.exceptions import InitialisationError
+from virtual_rainforest.core.logger import LOGGER
+from virtual_rainforest.core.utils import check_valid_constant_names, set_layer_roles
+from virtual_rainforest.models.litter.constants import LitterConsts
+from virtual_rainforest.models.litter.litter_pools import calculate_litter_pool_updates
+
+
+class LitterModel(BaseModel):
+    """A class defining the litter model.
+
+    This model can be configured based on the data object and a config dictionary. At
+    present the underlying model this class wraps is quite simple (i.e. two litter
+    pools), but this will get more complex as the Virtual Rainforest develops.
+
+    Args:
+        data: The data object to be used in the model.
+        update_interval: Time to wait between updates of the model state.
+        soil_layers: The number of soil layers to be modelled.
+        canopy_layers: The number of canopy layers to be modelled.
+        constants: Set of constants for the litter model.
+    """
+
+    model_name = "litter"
+    """An internal name used to register the model and schema"""
+    lower_bound_on_time_scale = "30 minutes"
+    """Shortest time scale that the litter model can sensibly capture."""
+    upper_bound_on_time_scale = "3 months"
+    """Longest time scale that the litter model can sensibly capture."""
+    required_init_vars = (
+        ("litter_pool_above_metabolic", ("spatial",)),
+        ("litter_pool_above_structural", ("spatial",)),
+    )
+    """Required initialisation variables for the litter model.
+
+    This is a set of variables that must be present in the data object used to create a
+    LitterModel , along with any core axes that those variables must map on to."""
+    vars_updated = ["litter_pool_above_metabolic", "litter_pool_above_structural"]
+    """Variables updated by the litter model."""
+
+    def __init__(
+        self,
+        data: Data,
+        update_interval: Quantity,
+        soil_layers: int,
+        canopy_layers: int,
+        constants: LitterConsts,
+        **kwargs: Any,
+    ):
+        super().__init__(data, update_interval, **kwargs)
+
+        # Check that litter pool data is appropriately bounded
+        if np.any(data["litter_pool_above_metabolic"] < 0.0) or np.any(
+            data["litter_pool_above_structural"] < 0.0
+        ):
+            to_raise = InitialisationError(
+                "Initial litter pools contain at least one negative value!"
+            )
+            LOGGER.error(to_raise)
+            raise to_raise
+
+        self.data
+        """A Data instance providing access to the shared simulation data."""
+        self.update_interval
+        """The time interval between model updates."""
```

Comment:
Type hints?

Reply:
I actually think these are redundant anyway. Like both are supplied to the `super().__init__()` above so I'm not sure why they are even included here (type hints for both come through the arguments to this `__init__` function

Reply:
True.

```suggestion
```